internal/pkg/cache/redis/redis.go

```diff
@@ -39,22 +42,42 @@ func NewRedis(address string) *Redis {
 
 // NewRedisWithConfig creates a new Redis instance with configurable options.
 func NewRedisWithConfig(cfg *config.RedisConfig) *Redis {
-	o := &redis.Options{
-		Addr:         cfg.Address,
-		MinIdleConns: cfg.MinIdleConns,
-		PoolSize:     cfg.PoolSize,
-		IdleTimeout:  cfg.IdleTimeout,
-		MaxConnAge:   cfg.MaxConnAge,
+	o := &redis.UniversalOptions{
+		Addrs:           parseRedisAddress(cfg.Address),
+		MinIdleConns:    cfg.MinIdleConns,
+		PoolSize:        cfg.PoolSize,
+		ConnMaxIdleTime: cfg.IdleTimeout,
```

Comment:
IdleTimeout and MaxConnAge have been replaced by ConnMaxIdleTime and ConnMaxLifetime.